src/sprites/setup.py

```diff
@@ -0,0 +1,257 @@
+import os
+from dataclasses import dataclass
+
+import pygame
+
+from src.enums import Direction, EntityState
+from src.settings import SCALE_FACTOR, CHAR_TILE_SIZE
+from src.support import resource_path
+
+
+@dataclass
+class _AniFrames:
+    frames: list[pygame.Surface]
+    hitbox: pygame.Rect
+
+    def get_frame(self, index: int) -> pygame.Surface:
+        return self.frames[int(index % len(self.frames))]
+
+    def get_hitbox(self) -> pygame.Rect:
+        return self.hitbox
+
+    def __len__(self) -> int:
+        return len(self.frames)
+
+
+@dataclass
+class _EntityStateAsset:
+    _data: dict[Direction, _AniFrames]
+
+    def __getitem__(self, direction: Direction) -> _AniFrames:
+        return self._data[direction]
```

Comment:
Actually, I don't think this is used anywhere...

Reply:
Moreover it could pretty much be replaced with a simple type alias, due to how you structured it

Reply:
It seems like I wanted to start abstracting the _EntityState class even further, though you're absolutely right. Changed it to a simple type alias 😄